charts/pomerium/templates/_helpers.tpl

```diff
@@ -210,6 +210,20 @@ Adapted from : https://github.com/helm/charts/blob/master/stable/drone/templates
 {{- end -}}
 {{- end -}}
 
+{{/* Determine secret name for Proxy JWT signing key */}}
+{{- define "pomerium.proxy.jwtSecret.name" -}}
```

Comment:
@tgomas I wonder if it wouldn't be best to keep the same naming as the variable and the configuration, i.e., `signingKey` without using the terms `jwt` for example. `pomerium.signingKeySecret.name`.

Same for the the values: `proxy.signingKey` and `proxy.existingSigningKeySecret` seem good.

Reply:
(and same for the names below, jwt shouldn't appear in the name IMHO)

Reply:
I introduced `jwt` in naming to highlight the difference with `tls` secrets since `signingKey` would not be clear in a context where everything is related to TLS.

It could make sense to extract secret's definition in `signing-key-secret.yaml` and rename everything with `signingKey` instead `jwt`. In that case, should the secret also require dedicated values to switch its generation on since it will no more be related to tls secrets ?

Reply:
I think the fact the generated signing key is a gpg key is a different matter from the fact that a tls secret (which is a special kind of secret in k8s) is used.

IMHO:
- the signing key should be in a normal secret resource
- the signing key should be generated in the same manner as tls secrets resource
- the signing key could/should be stored with cookie and shared secrets in the same secret resource

Reply:
@travisgroth what do you think about that ?

It's different from [your first advice](https://github.com/pomerium/pomerium-helm/pull/66#discussion_r389578953) but I can make changes in that way.

Reply:
re: naming, I'm partial to sticking close to the parameter name in pomerium which is just `SIGNING_KEY`, for better or worse.  No hard opinion here though.  

>the signing key should be in a normal secret resource

I was actually thinking a TLS secret.  Isn't it just another x509 keypair?  Operationally it is unimportant either way, though.  

>the signing key should be generated in the same manner as tls secrets resource

Yes

>the signing key could/should be stored with cookie and shared secrets in the same secret resource

This won't work with hooks, unfortunately.  It will need to be its own object to be controlled correctly.

Reply:
Signing key is only a key and not a keypair. Therefore it is now store in dedicated Opaque secret